src/averageAges.js

```diff
@@ -37,7 +46,17 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const mothers = people.filter(person => person.mother !== null)
+    .map(person => person.mother);
+
+  const women = people.filter(person => withChildren
+    ? mothers.includes(person.name)
+    : person.sex === 'f'
+  );
```

Comment:
зайву роботу знову робиш:)
```suggestion
  const women = people.filter(woman => withChildren
    ? people.find(child => child.mother === woman.name)
    : person.sex === 'f'
  );
```
шукаємо жінок - якщо з дітьми - то шукаємо в таблиці в кого ця людина записана як мама (тобто вона 100% жінка), якщо без дітей, то просто всіх жінок

Reply:
Спасибо за исправления, приятно довести дело до конца)